Azure.ResourceManager.Core/Resources/ResourceType.cs

```diff
@@ -87,7 +87,7 @@ public ResourceType Parent
 
         public int CompareTo(ResourceType other)
         {
-            if (other == null)
+            if ((object)other == null)
```

Comment:
typically you use reference equals here instead of a cast
```suggestion
            if (ReferenceEquals(other, null))
```